securitycenter/bigqueryv2/bigquery_test.go

```diff
@@ -0,0 +1,297 @@
+	"cloud.google.com/go/securitycenter/apiv2/securitycenterpb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"github.com/google/uuid"
+)
+
+func orgID(t *testing.T) string {
+	orgID := os.Getenv("GCLOUD_ORGANIZATION")
+	if orgID == "" {
+		t.Skip("GCLOUD_ORGANIZATION not set")
+	}
+	return orgID
+}
+
+func bigQueryDatasetName(t *testing.T) string {
+	bigQueryDatasetName := os.Getenv("SCC_BIGQUERY_DATASET_NAME")
+	if bigQueryDatasetName == "" {
+		t.Skip("SCC_BIGQUERY_DATASET_NAME not set")
+	}
+	return bigQueryDatasetName
+}
+
+func addBigQueryExport(t *testing.T, bigQueryExportID string) error {
+	orgID := orgID(t)
+	bigQueryDatasetName := bigQueryDatasetName(t)
+
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("securitycenter.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	bigQueryExport := &securitycenterpb.BigQueryExport{
+		Description: "BigQueryExport that receives all HIGH severity Findings",
+		Filter:      "severity=\"HIGH\"",
+		Dataset:     bigQueryDatasetName,
+	}
+
+	req := &securitycenterpb.CreateBigQueryExportRequest{
+		Parent:           fmt.Sprintf("organizations/%s/locations/global", orgID),
+		BigQueryExport:   bigQueryExport,
+		BigQueryExportId: bigQueryExportID,
+	}
+
+	_, err0 := client.CreateBigQueryExport(ctx, req)
+	if err0 != nil {
+		return fmt.Errorf("Failed to create BigQueryConfig: %w", err)
+	}
+	return nil
+}
+
+func cleanupBigQueryExport(t *testing.T, bigQueryExportID string) error {
+	orgID := orgID(t)
+
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+
+	if err != nil {
+		return fmt.Errorf("securitycenter.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	req := &securitycenterpb.DeleteBigQueryExportRequest{
+		Name: fmt.Sprintf("organizations/%s/locations/global/bigQueryExports/%s", orgID, bigQueryExportID),
+	}
+
+	if err := client.DeleteBigQueryExport(ctx, req); err != nil {
+		return fmt.Errorf("failed to delete BigQueryExport: %w", err)
+	}
+
+	return nil
+}
+
+func TestListBigQuery(t *testing.T) {
+
+	testutil.Retry(t, 5, 20*time.Second, func(r *testutil.R) {
+		buf := new(bytes.Buffer)
+
+		// Create Test BigQueryExport Config
+		rand, err := uuid.NewUUID()
+		if err != nil {
+			r.Errorf("Issue generating id.")
+			return
+		}
+		configID := "random-bqexport-id-" + rand.String()
+
+		if err := addBigQueryExport(t, configID); err != nil {
+			r.Errorf("Could not setup test environment: %v", err)
+			return
+		}
+
+		parent := fmt.Sprintf("organizations/%s/locations/global", orgID(t))
+
+		// Call List BigQueryExport
+		err = listBigQueryExport(buf, parent)
+
+		if err != nil {
+			r.Errorf("listBigQueryExport() had error: %v", err)
+			return
+		}
+
+		got := buf.String()
+
+		fmt.Println(got)
+
+		if !strings.Contains(got, configID) {
+			r.Errorf("listBigQueryConfigs() got: %s want %s", got, configID)
+		}
+
+		// Cleanup
+		cleanupBigQueryExport(t, configID)
```

Comment:
These cleanups are ok as they are, since the tests are pretty straightforward.

For the future, you may want to use [`defer`](https://go.dev/tour/flowcontrol/12) or [`t.Cleanup`](https://pkg.go.dev/testing#T.Cleanup) to ensure the cleanup function is always called, no matter how the function exits.